File: android/app/src/main/java/com/github/bfollon/farmaciasdeguardiaensegovia/ui/components/NextShiftCard.kt
/*
 * Copyright (C) 2025  Bruno Follon (@bFollon)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.github.bfollon.farmaciasdeguardiaensegovia.ui.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccessTime
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import com.github.bfollon.farmaciasdeguardiaensegovia.data.DutyTimeSpan
import com.github.bfollon.farmaciasdeguardiaensegovia.data.Pharmacy

/**
 * Compact component for displaying the next shift information
 * Shows "Siguiente turno" label with pharmacy name(s)
 * Tap to open detailed modal with full shift and pharmacy information
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun NextShiftCard(
    timeSpan: DutyTimeSpan,
    pharmacies: List<Pharmacy>,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Card(
        onClick = onClick,
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceVariant
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp),
        shape = RoundedCornerShape(12.dp)
    ) {
        Row(
            modifier = Modifier.padding(12.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Icon(
                imageVector = Icons.Default.AccessTime,
                contentDescription = "Siguiente turno",
                tint = MaterialTheme.colorScheme.secondary,
                modifier = Modifier.size(20.dp)
            )

            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = "Siguiente turno",
                    style = MaterialTheme.typography.labelMedium,
                    color = MaterialTheme.colorScheme.secondary,
                    fontWeight = FontWeight.Medium
                )

                // Pharmacy name(s) - show first pharmacy, "+X más" if multiple
                val displayText = if (pharmacies.size == 1) {
                    pharmacies[0].name
                } else {
                    "${pharmacies[0].name} +${pharmacies.size - 1} más"
                }

                Text(
                    text = displayText,
                    style = MaterialTheme.typography.bodyMedium,
                    fontWeight = FontWeight.SemiBold,
                    color = MaterialTheme.colorScheme.onSurface,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis
                )
            }

            // Chevron icon to indicate expandability
            Icon(
                imageVector = Icons.Default.ChevronRight,
                contentDescription = "Ver detalles",
                tint = MaterialTheme.colorScheme.outline,
                modifier = Modifier.size(20.dp)
            )
        }
    }
}

File: ios/FarmaciasDeGuardiaEnSegovia/Views/NextShiftCard.swift
/*
 * Copyright (C) 2025  Bruno Follon (@bFollon)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import SwiftUI

/// Card displaying information about the next pharmacy shift
struct NextShiftCard: View {
    let schedule: PharmacySchedule
    let timeSpan: DutyTimeSpan
    let region: Region
    @State private var showingDetails = false

    private var pharmacies: [Pharmacy] {
        schedule.shifts[timeSpan] ?? []
    }

    private var shiftLabel: String {
        switch timeSpan {
        case .capitalDay:
            return "Diurno"
        case .capitalNight:
            return "Nocturno"
        case .fullDay:
            return "24 horas"
        case .ruralDaytime, .ruralExtendedDaytime:
            return "Diurno"
        default:
            return "Turno"
        }
    }

    private var sheetHeight: CGFloat {
        let baseHeight: CGFloat = 150  // Nav bar + inline shift header + padding
        let pharmacyHeight: CGFloat = 120  // Height per pharmacy including spacing
        let pharmacyCount = CGFloat(pharmacies.count)
        return baseHeight + (pharmacyCount * pharmacyHeight)
    }

    var body: some View {
        Button {
            showingDetails = true
        } label: {
            HStack(alignment: .center, spacing: 12) {
                // Left side - all content in VStack
                VStack(alignment: .leading, spacing: 12) {
                    // Header
                    HStack(spacing: 8) {
                        Image(systemName: "clock.fill")
                            .foregroundColor(.secondary)
                            .frame(width: 20)
                        Text("Siguiente turno")
                            .font(.subheadline)
                            .foregroundColor(.secondary)
                    }

                    // Shift info
                    HStack(spacing: 8) {
                        Image(systemName: timeSpan == .capitalNight ? "moon.stars.fill" : "sun.max.fill")
                            .foregroundColor(.secondary.opacity(0.7))
                            .frame(width: ViewConstants.iconColumnWidth)
                        Text("\(shiftLabel) • \(timeSpan.displayName)")
                            .font(.headline)
                            .foregroundColor(.primary)
                    }

                    // Compact pharmacy preview
                    if let firstPharmacy = pharmacies.first {
                        HStack(spacing: 8) {
                            Text(firstPharmacy.name)
                                .font(.body)
                                .foregroundColor(.primary)
                                .lineLimit(1)

                            if pharmacies.count > 1 {
                                Text("+\(pharmacies.count - 1) más")
                                    .font(.caption)
                                    .foregroundColor(.secondary)
                            }
                        }
                    }
                }

                Spacer()

                // Right side - chevron
                Image(systemName: "chevron.right")
                    .foregroundColor(.secondary)
                    .font(.system(size: 14))
            }
            .padding()
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(
                RoundedRectangle(cornerRadius: 8)
                    .fill(Color.gray.opacity(0.1))
            )
            .contentShape(RoundedRectangle(cornerRadius: 8))
        }
        .buttonStyle(.plain)
        .sheet(isPresented: $showingDetails) {
            NextShiftDetailSheet(
                schedule: schedule,
                timeSpan: timeSpan,
                region: region
            )
            .presentationDetents([.height(sheetHeight)])
            .presentationDragIndicator(.visible)
        }
    }
}
